src/sentry/tasks/integrations/migrate_issues.py

```diff
@@ -0,0 +1,109 @@
+from django.db import IntegrityError, transaction
+
+from sentry.models import (
+    ExternalIssue,
+    Group,
+    GroupLink,
+    GroupMeta,
+    Integration,
+    OrganizationIntegration,
+    Project,
+)
+from sentry.plugins.base import plugins
+from sentry.tasks.base import instrumented_task, retry
+from sentry.tasks.integrations import logger
+
+
+@instrumented_task(
+    name="sentry.tasks.integrations.migrate_issues",
+    queue="integrations",
+    default_retry_delay=60 * 5,
+    max_retries=5,
+)
+@retry(exclude=(Integration.DoesNotExist))
+def migrate_issues(integration_id: int, organization_id: int) -> None:
+    from sentry_plugins.jira.plugin import JiraPlugin
+
+    integration = Integration.objects.get(id=integration_id)
+    for project in Project.objects.filter(organization_id=organization_id):
+        plugin = None
+        for p in plugins.for_project(project):
+            if isinstance(p, JiraPlugin) and p.is_configured(None, project):
+                plugin = p
+                break
+
+        if not plugin:
+            continue
+
+        is_different_jira_instance = plugin.get_option("instance_url", project).rstrip(
+            "/"
+        ) != integration.metadata.get("base_url").rstrip("/")
+        if is_different_jira_instance:
+            continue
+        groups = Group.objects.filter(project=project.id)
+        # this seems expensive but I don't think there are other options
+        plugin_issues = GroupMeta.objects.filter(
+            key=f"{plugin.slug}:tid", group__id__in=[group.id for group in groups]
+        )
+        for plugin_issue in plugin_issues:
+            external_issue, _ = ExternalIssue.objects.get_or_create(
+                organization_id=organization_id,
+                integration_id=integration_id,
+                key=plugin_issue.value,
+            )
+            try:
+                with transaction.atomic():
+                    GroupLink.objects.create(
+                        group_id=plugin_issue.group_id,
+                        project_id=project.id,
+                        linked_type=GroupLink.LinkedType.issue,
+                        linked_id=external_issue.id,
+                        relationship=GroupLink.Relationship.references,
+                    )
+            except IntegrityError:
+                continue
+
+            plugin_issue.delete()
```

Comment:
Do you want to delete the plugin_issue even if the GroupLink fails to be created?

Reply:
That is a good question. I think so, because we'll still disable the plugin which hides the linked issues from the sidebar anyway. If we kept them around they're not useful to the user anyway. What do you think?

Reply:
If we're going to disable the plugin anyways then I agree there is no point in keeping the groupmeta records around.